source/_posts/2015-09-03-hiring-apprentices.markdown

```diff
@@ -0,0 +1,82 @@
+---
+layout: post
+title: "Hiring Apprentices"
+date: 2015-09-09 14:44:39 -0600
+comments: true
+categories: hiring, apprenticeship, mentoring, ruby
+---
+
+The [MX](http://mx.com/) apprenticeship has been a lot of work and fun so far.
+This blog post is the beginning of an effort to document what I've learned along the way.
+
+# The Goal
+
+Initially I got interested in the idea of an apprenticeship while listening to the [ruby rogues episode with Joe Mastey and Jill Lynch](https://devchat.tv/ruby-rogues/190-rr-apprenticeship-with-joseph-mastey-and-jill-lynch-of-enova).
+I help to organize a [local ruby meetup group](http://utruby.org/#uv.rb) so I liked the idea of helping new engineers.
+I also love my job at MX so I liked the idea of helping to grow our team.
+
+When I discussed the idea of an apprenticeship with people at MX, the first thing I had to do was nail down what our objective was. "Helping new engineers" wasn't going to cut it.
+
+We decided that the primary objective was to __add amazing engineers to the team__.
+
+# Amazing Apprentices
+
+Based on my experience in the local Ruby community I believed that there were some really amazing programmers who were self-taught or graduating from bootcamps that were struggling to find first jobs.
+
+The shortage of senior engineers at pretty much every company has made it really difficult to get a job as a junior developer. Lots of companies think that they can't afford to hire junior engineers because they don't have enough senior people to train them.
+There are also serious bias issues throughout the software industry and especially in the hiring processes.
+
+So we had a hypothesis that there was a large pool of extremely talented and hard-working people that was underutilized.
+
+# Teaching
+
+On our team we usually only considered hiring someone if they had a full year of professional programming experience.
+This is partially because our team has no managers and no task assignment so it is easy to get lost in the weeds if you don't have your own discipline for staying focused as an engineer.
+But this meant that if we wanted to make an apprenticeship work for us, we needed  the apprentices to have a skill and experience level similar to someone with one full year of professional experience.
+
+But I didn't think it should actually take a year to gain that experience and those skills.
+
+It is all too common for people to have a first programming job where they get assigned a laundry list of bugs and work alone [spending lots of time](http://www.happybearsoftware.com/how-to-survive-your-first-year-as-a-programmer.html) on just a few small conceptual misunderstandings.
```

Comment:
"...laundry list of bugs and work alone." This phrase is a little confusing and can be rearranged I think.

Perhaps... "...they work alone and are assigned a laundry list of bugs, spending lots of time..."